tests/test/discriminant_kind.rs

```diff
@@ -0,0 +1,138 @@
+use super::*;
```

Comment:
Can you write some comments here for what each of the test goals are supposed to be for?